remote-wallet/src/ledger.rs

```diff
@@ -227,7 +227,7 @@ impl LedgerWallet {
         data: &[u8],
     ) -> Result<Vec<u8>, RemoteWalletError> {
         self.write(command, p1, p2, data)?;
-        if p1 == P1_CONFIRM {
+        if p1 == P1_CONFIRM && (p2 == 0 || p2 == P2_EXTEND) {
```

Comment:
This needs a comment. Your intent isn't clear. `p2 == P2_EXTEND` means:
* `(p2 & P2_EXTEND) && (p2 & P2_MORE == 0)` aka, the last of a multi-part message
* No other bits in p2 will ever be used in the future

Why not this?

```rust
// If this is the first of a multi-part message
if (p1 == P1_CONFIRM) && ((p2 & P2_EXTEND) == 0) {
```

Reply:
It doesn't seem quite correct to print the message during the first chunk of a multi-part message. If something goes wrong with subsequent sends, the the cli may not in fact be waiting for an action from the device.

Could we indicate the last of a multi-part message more categorically?

Reply:
How about this, which only assumes that the 2nd bit won't be used for anything else in the future?